File: typespec/common/localization.go
package common

import (
	"strings"

	"github.com/moov-io/iso3166"
)

type CountryCode string

type DisplayName string

func IsCountryCode(value CountryCode) bool {
	code := string(value)
	return code == strings.ToUpper(code) && iso3166.ValidAlpha2(code)
}

func NormalizeDisplayName(value DisplayName) DisplayName {
	return DisplayName(strings.TrimSpace(string(value)))
}

func IsDisplayName(value DisplayName) bool {
	length := len([]rune(NormalizeDisplayName(value)))
	return length >= 1 && length <= 200
}

File: typespec/common/localization.ts
export const frontendLocaleValues = ["en-US", "ta", "de-DE"] as const;

export type FrontendLocale = (typeof frontendLocaleValues)[number];

const frontendLocales = new Set<string>(frontendLocaleValues);

export type CountryCode = string;

export type DisplayName = string;

export function isFrontendLocale(value: unknown): value is FrontendLocale {
  return typeof value === "string" && frontendLocales.has(value);
}

export function normalizeDisplayName(value: DisplayName): DisplayName {
  return value.trim();
}

export function isDisplayName(value: DisplayName): boolean {
  const length = [...normalizeDisplayName(value)].length;
  return length >= 1 && length <= 200;
}
